src/material/stepper/_stepper-theme.scss

```diff
@@ -17,10 +17,14 @@
   .mat-step-header {
     &.cdk-keyboard-focused,
     &.cdk-program-focused,
-    &:hover {
+    &:hover:not([aria-disabled]) {
```

Comment:
This will cover `aria-disabled="false"` as well. Should the selector be more specific?

Reply:
Right now we only add the `aria-disabled` attribute when its equal to true. Although I think being more specific makes sense in case that ever changes